central/processlisteningonport/datastore/datastore_impl.go

```diff
@@ -0,0 +1,503 @@
+package datastore
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/stackrox/rox/central/metrics"
+	processIndicatorStore "github.com/stackrox/rox/central/processindicator/datastore"
+	"github.com/stackrox/rox/central/processlisteningonport/store/postgres"
+	"github.com/stackrox/rox/central/role/resources"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+type datastoreImpl struct {
+	storage            postgres.Store
+	indicatorDataStore processIndicatorStore.DataStore
+}
+
+var (
+	plopSAC = sac.ForResource(resources.DeploymentExtension)
+	log     = logging.LoggerForModule()
+)
+
+func newDatastoreImpl(
+	storage postgres.Store,
+	indicatorDataStore processIndicatorStore.DataStore,
+) *datastoreImpl {
+	return &datastoreImpl{
+		storage:            storage,
+		indicatorDataStore: indicatorDataStore,
+	}
+}
+
+func (ds *datastoreImpl) AddProcessListeningOnPort(
+	ctx context.Context,
+	portProcesses ...*storage.ProcessListeningOnPortFromSensor,
+) error {
+	defer metrics.SetDatastoreFunctionDuration(
+		time.Now(),
+		"ProcessListeningOnPort",
+		"AddProcessListeningOnPort",
+	)
+
+	if !env.PostgresDatastoreEnabled.BooleanSetting() {
+		// PLOP is a Postgres-only feature, do nothing.
+		log.Warnf("Tried to add PLOP not on Postgres, ignore: %+v", portProcesses)
+		return nil
+	}
+
+	if ok, err := plopSAC.WriteAllowed(ctx); err != nil {
+		return err
+	} else if !ok {
+		return sac.ErrResourceAccessDenied
+	}
+
+	portProcesses, completedInBatch := normalizePLOPs(portProcesses)
```

Comment:
Do we really want to reuse the name of the incoming parameter?  I would think we would want to name it something else to differentiate since we pass that into `normalizePLOPs` and get 2 things back.

Reply:
No. portProcesses has been changed to normalizedPLOPs